### PR TITLE
Added support for Scala.js

### PR DESCRIPTION
**Pull request by mpilquist**
This PR adds support for Scala.js, including SBT runner support. I had to drop Scala 2.10 from the build matrix as Scala.js is no longer supporting 2.10.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ src_managed/
 project/boot/
 project/plugins/project/
 project/source.scala
+.bloop
```
Comment (charleso) [inline .gitignore]: Super nit, but any reason not to just put these in your user/global ignore list instead of manually adding them to the repo itself?

**Comment by charleso** [comment]
Looks pretty low-impact to me (I've never used Scala.js). 

@steinybot Any thoughts/concerns?

Otherwise I'm happy to merge.

**Comment by charleso** [comment]
Nice work btw!

File: .gitignore (hunk 1 above)
Comment (mpilquist) [reply .gitignore]: I always put them in repo gitignores due to work on projects with lots of committers who all have different user/global ignore lists. One less thing to deal with in PR reviews. :) I don't feel strongly about this though and I'm happy to remove the changes to .gitignore here.

**Pull request merged**